src/nvim/ops.c

```diff
@@ -2521,62 +1790,7 @@ int op_change(oparg_T *oap)
   return retval;
 }
 
-#if defined(EXITFREE)
-void clear_registers(void)
-{
-  for (int i = 0; i < NUM_REGISTERS; i++) {
-    free_register(&y_regs[i]);
-  }
-}
-
-#endif
-
-/// Free contents of yankreg `reg`.
-/// Called for normal freeing and in case of error.
-///
-/// @param reg  must not be NULL (but `reg->y_array` might be)
-void free_register(yankreg_T *reg)
-  FUNC_ATTR_NONNULL_ALL
-{
-  XFREE_CLEAR(reg->additional_data);
-  if (reg->y_array == NULL) {
-    return;
-  }
-
-  for (size_t i = reg->y_size; i-- > 0;) {  // from y_size - 1 to 0 included
-    API_CLEAR_STRING(reg->y_array[i]);
-  }
-  XFREE_CLEAR(reg->y_array);
-}
-
-/// Yanks the text between "oap->start" and "oap->end" into a yank register.
-/// If we are to append (uppercase register), we first yank into a new yank
-/// register and then concatenate the old and the new one.
-/// Do not call this from a delete operation. Use op_yank_reg() instead.
-///
-/// @param oap operator arguments
-/// @param message show message when more than `&report` lines are yanked.
-/// @returns whether the operation register was writable.
-bool op_yank(oparg_T *oap, bool message)
-  FUNC_ATTR_NONNULL_ALL
-{
-  // check for read-only register
-  if (oap->regname != 0 && !valid_yank_reg(oap->regname, true)) {
-    beep_flush();
-    return false;
-  }
-  if (oap->regname == '_') {
-    return true;  // black hole: nothing to do
-  }
-
-  yankreg_T *reg = get_yank_register(oap->regname, YREG_YANK);
-  op_yank_reg(oap, message, reg, is_append_register(oap->regname));
-  set_clipboard(oap->regname, reg);
-  do_autocmd_textyankpost(oap, reg);
-  return true;
-}
-
-static void op_yank_reg(oparg_T *oap, bool message, yankreg_T *reg, bool append)
+void op_yank_reg(oparg_T *oap, bool message, yankreg_T *reg, bool append)
```

Comment:
Most of `op_yank()` is moved to `op_yank_reg()` in Nvim, so `op_yank_reg()` should be moved to register.c. And patch 8.1.2077 moved `yank_copy_line()` (which is only called by `op_yank()` in Vim or `op_yank_reg()` in Nvim) as well, but isn't moved in this PR.